kafka.go

```diff
@@ -340,15 +367,18 @@ func (client *KafkaClient) runCallback(event *Event, consumerMessage kafka.Messa
 		return
 	}
 
-	callback, ok := innerValue.(func(*Event, error))
+	callback, ok := innerValue.(func(context.Context, *Event, error))
 	if !ok {
 		log.Error("unable to convert interface to callback function")
 		return
 	}
 
+	ctx, ctxClose := context.WithCancel(context.Background())
+	defer ctxClose()
```

Comment:
it's better to pass a context from the `subscribeBuilder` . it could be done if we would store an instance of the  `subscribeBuilder` in the `callbackMap`

Reply:
does it mean that we will have another sync map to save the context, just like the callbackMap? I will adjust it that way. CMIIW